data/resources/ui/login-code.blp

```diff
@@ -0,0 +1,69 @@
+using Gtk 4.0;
+using Adw 1;
+
+template LoginCode {
+  layout-manager: BoxLayout {
+    orientation: vertical;
+  };
+  vexpand: true;
+
+  Adw.HeaderBar {
+    styles ["flat"]
+
+    [start]
+    Button back_button {
+      action-name: "login.previous";
+      icon-name: "go-previous-symbolic";
+    }
+  }
+
+  WindowHandle {
```

Comment:
Not tested, but I'm pretty sure that will not work with touchscreen because the `AdwStatusPage` under it is scrollable. In general scrollable content are not draggable like that, so I guess you can remove it. BTW `AdwToolbarView` now does all that work of putting `GtkWindowHandle` where necessary automatically, so you generally don't need to worry about anymore.

Reply:
Yeah..., didn't consider this. I'll remove it.